### PR TITLE
Add default affinity and topologySpreadConstraints

### PR DESCRIPTION
**Pull request by ubergesundheit**
Towards https://github.com/giantswarm/giantswarm/issues/21426

This PR adds `affinity` and `topologySpreadConstraints` to make sure pods are spread evenly across zones.

This PR...

## Checklist

- [x] Automated test are working
- [x] Changelog entry has been added

### Manual tests on workload clusters

Played around with multiple zones, changing number of replicas, changing number of zones

### CODE REVIEW

**Comment by ubergesundheit** [comment]
@webwurst you may alreay look, but lets try this one on clusters to make sure everything works

**Comment by ubergesundheit** [comment]
The kind cluster does not have the node labels.. the pod shows

```
Warning  FailedScheduling  36s (x6 over 3m52s)  default-scheduler  0/1 nodes are available: 1 node(s) didn't match pod topology spread constraints (missing required label).
```

I've added it to the kind config, but maybe that this was just a sign that the affinity does not work?

File: helm/kong-app/values.yaml [hunk 1]
```diff
@@ -611,11 +611,34 @@ terminationGracePeriodSeconds: 30
...
+affinity:
+  nodeAffinity:
+    requiredDuringSchedulingIgnoredDuringExecution:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: kubernetes.io/role
+          operator: NotIn
+          values:
+          - master
+          - controlplane
```
Comment (ubergesundheit) [inline helm/kong-app/values.yaml:623]: We have to check these in the wild

Comment (webwurst) [inline helm/kong-app/values.yaml:623]: I found also this: https://github.com/kubernetes/enhancements/blob/master/keps/sig-cluster-lifecycle/kubeadm/2067-rename-master-label-taint/README.md#summary

**Comment by ubergesundheit** [comment]
This probably needs testing on kvm

**Comment by ubergesundheit** [comment]
> This probably needs testing on kvm

My suspicions were right: kvm nodes also don't have the `zone` label so nothing gets scheduled.

**Comment by ubergesundheit** [comment]
@webwurst please review the documentation changes :) thank you

File: helm/kong-app/templates/_helpers.tpl [hunk 2]
```diff
@@ -25,13 +25,22 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 
+{{- define "kong.imagesHash" -}}
+{{- if .Values.image.registry }}
```
Comment (webwurst) [inline helm/kong-app/templates/_helpers.tpl:29]: Not sure, maybe?
```suggestion
{{- if .Values.image.registry -}}
```

File: helm/kong-app/templates/_helpers.tpl [hunk 3]
```diff
@@ -25,13 +25,22 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
...
+{{- printf "%s-%s-%s" .Values.image.registry (include "kong.getRepoTag" .Values.image) (include "kong.getRepoTag" .Values.ingressController.image) | sha256sum | trunc 63 -}}
+{{- else }}
```
Comment (webwurst) [inline helm/kong-app/templates/_helpers.tpl:31]: Same as above?
```suggestion
{{- else -}}
```

**Review by webwurst — APPROVED** [review]

File: helm/kong-app/templates/_helpers.tpl (hunk 2 above)
Comment (ubergesundheit) [reply helm/kong-app/templates/_helpers.tpl:29]: Tried locally, does not make any difference.

File: helm/kong-app/templates/_helpers.tpl (hunk 3 above)
Comment (ubergesundheit) [reply helm/kong-app/templates/_helpers.tpl:31]: Tried locally, does not make any difference.

**Pull request merged**